cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,248 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Pytest test configuration file.
+
+"""
+import pytest
+from pathlib import Path
+from subprocess import run, TimeoutExpired
+from time import sleep
+import json
+import os
+import shutil
+
+@pytest.fixture(scope="session")
+def installer_path(request):
+    return Path(request.config.getoption("--installer-path")).resolve()
+
+@pytest.fixture(scope="session")
+def install_root(request):
+    return Path(request.config.getoption("--install-root")).resolve()
+
+@pytest.fixture(scope="session")
+def project_path(request):
+    # MSBuild doesn't want us to output in the temp directory MSB8029
+    # so we use a pre-defined folder outside of %TMP%
+    #return tmpdir_factory.mktemp("TestProject")
+    path = Path(request.config.getoption("--project-path")).resolve()
+    yield path
+
+    # wait a few seconds for processes to quit
+    sleep(5)
+
+    shutil.rmtree(path)
+
+@pytest.fixture(scope="session")
+def cmake_path(install_root):
+    return install_root / 'cmake' / 'runtime' / 'cmake-3.20.2-windows-x86_64' / 'bin' / 'cmake.exe'
+
+@pytest.fixture(scope="session")
+def test_installer(installer_path, install_root):
+    assert installer_path.is_file(), f"Invalid installer path {installer_path}"
+
+    # when the installer is run with timeout of 30 minutes
+    result = run([installer_path , f"InstallFolder={install_root}", "/quiet"], timeout=30*60)
```

Comment:
Here and throughout- Are the subprocess logs important? Could potentially save the output in case something goes wrong to make debugging easier.

Reply:
The <build>/Testing folder automatically gets uploaded in AR so directing the output there would be optimal

Reply:
This particular test would not be part of AR, it is run separately and depends on the nightly installer build job

Reply:
Currently I have the subprocess logs printing so they will appear in the regular Jenkins logs.  It appears I need to do some other config changes to enable the UploadTestArtifacts stage that are not very clear from the code.  The current test does not use CTEST so I'm not sure what adding blank options would do when the engine is not getting built and no ctests are run on the node.
```
                        // Upload test artifacts only on builds that failed and ran test suites
                        if (env.IS_UPLOAD_TEST_ARTIFACTS?.toBoolean() && params.containsKey('CTEST_OPTIONS')) {
                            CreateUploadTestArtifactStage(build_job_name, envVars['WORKSPACE'], params.OUTPUT_DIRECTORY).call()
                        }
```